tests/sentry/coreapi/tests.py

```diff
@@ -441,6 +441,16 @@ def test_environment_too_long(self):
         assert data['errors'][0]['name'] == 'environment'
         assert data['errors'][0]['value'] == 'a' * 65
 
+    def test_environment_invalid(self):
+        data = self.validate_and_normalize({
+            'environment': 'a/b',
+        })
+        assert not data.get('environment')
+        assert len(data['errors']) == 1
+        assert data['errors'][0]['type'] == 'invalid_data'
```

Comment:
What are your thoughts on adding a new error in `src/sentry/models/eventerror.py`, something like `INVALID_ENVIRONMENT`, with a user-visible string that says something about the fact that we no longer allow slashes and stuff in here?

I'm just thinking that might be the best way to communicate the reason for this change to the ~100 or so customers who will now suddenly start seeing event processing errors in the UI and wonder why, since we don't really document the required pattern for these things in our docs (that I can find)

Reply:
Sure. That would be better for those folks over what they would get now.

Reply:
I've added a more specific error message when handling invalid environments. It looks like 

![screen shot 2018-10-03 at 3 07 35 pm](https://user-images.githubusercontent.com/24086/46433146-a03cb400-c73f-11e8-915b-6b5bd9533da9.png)

Reply:
💯